manifest-build-tools/docker_build.sh

```diff
@@ -37,51 +61,62 @@ doBuild() {
             exit 1
         fi
         pushd $repo
-            PKG_VERSION=""
+            PKG_TAG=""
             if [ "$BUILD_LATEST" != true ]; then
-                #use the provided version number if exists in .version
-                PKG_VERSION=$(source ./.version && PKG_VERSION=${PKG_VERSION//\~/-} && echo $PKG_VERSION)
-                VERSION=:${PKG_VERSION:=$DEFAULT_VERSION}
+                tagCalculate $repo
+                TAG=:${PKG_TAG}
             fi
-            echo "Building rackhd/$repo$VERSION"
-            repos_versions=$repos_versions$repo$VERSION" "
+            echo "Building rackhd/$repo$TAG"
+            repos_tags=$repos_tags$repo$TAG" "
             cp Dockerfile ../Dockerfile.bak
             if [ "$repo" != "on-core" ];then
                     # Use the new sources.list
-                    sed -i "/^FROM/a RUN mv /etc/apt/sources.list /etc/apt/sources.list.bak\nCOPY $SOURCE_LIST /etc/apt/sources.list" Dockerfile
+                    sed -i "/^FROM/a RUN mv /etc/apt/sources.list /etc/apt/sources.list.bak\nADD $SOURCE_LIST /etc/apt/sources.list" Dockerfile
                     #Based on newly build upstream image to build
-                    sed -i "/^FROM/ s/$/${PRE_VERSION}/" Dockerfile
+                    sed -i "/^FROM/ s/$/${PRE_TAG}/" Dockerfile
                     # Recover the sources.list
                     sed -i -e "\$aRUN mv /etc/apt/sources.list.bak /etc/apt/sources.list" Dockerfile
-                    docker build -t rackhd/$repo$VERSION .
+                    docker build -t rackhd/$repo$TAG .
             fi
             case $repo in
                 "on-core")
-                    docker build -t rackhd/$repo$VERSION .
-                    PRE_VERSION=$VERSION
+                    docker build -t rackhd/$repo$TAG .
+                    PRE_TAG=$TAG
                     ;;
                 "on-tasks")
-                    PRE_VERSION=$VERSION
+                    PRE_TAG=$TAG
                     ;;
             esac 
             mv ../Dockerfile.bak Dockerfile
```

Comment:
Each build will change original Dockerfile, add tag information, sourcelist ...etc. 
When build ```latest```, ```doBuild``` will run twice so maintain a clean Dockerfile is necessary.